packages/preset-uno/src/rules/animation.ts

```diff
@@ -2,15 +2,114 @@ import { Rule } from '@unocss/core'
+  'fade-out-up': '@keyframes fade-out-up {from {opacity:1;} to {opacity:0; transform:translate3d(0,-100%,0);}}',
+  'fade-out-up-big': '@keyframes fade-out-up-big {from {opacity:1;} to {opacity:0; transform:translate3d(0,-2000px,0);}}',
+  'fade-out-top-left': '@keyframes fade-out-top-left {from {opacity:1; transform:translate3d(0,0,0);} to {opacity:0; transform:translate3d(-100%,-100%,0);}}',
+  'fade-out-top-right': '@keyframes fade-out-top-right {from {opacity:1; transform:translate3d(0,0,0);} to {opacity:0; transform:translate3d(100%,-100%,0);}}',
+  'fade-out-bottom-left': '@keyframes fade-out-bottom-left {from {opacity:1; transform:translate3d(0,0,0);} to {opacity:0; transform:translate3d(-100%,100%,0);}}',
+  'fade-out-bottom-right': '@keyframes fade-out-bottom-right {from {opacity:1; transform:translate3d(0,0,0);} to {opacity:0; transform:translate3d(100%,100%,0);}}',
+  'back-in-up': '@keyframes back-in-up {0% {opacity:0.7; transform:translateY(1200px) scale(0.7);} 80% {opacity:0.7; transform:translateY(0px) scale(0.7);} 100% {opacity:1; transform:scale(1);}}',
+  'back-in-down': '@keyframes back-in-down {0% {opacity:0.7; transform:translateY(-1200px) scale(0.7);} 80% {opacity:0.7; transform:translateY(0px) scale(0.7);} 100% {opacity:1; transform:scale(1);}}',
+  'back-in-right': '@keyframes back-in-right {0% {opacity:0.7; transform:translateX(2000px) scale(0.7);} 80% {opacity:0.7; transform:translateY(0px) scale(0.7);} 100% {opacity:1; transform:scale(1);}}',
+  'back-in-left': '@keyframes back-in-left {0% {opacity:0.7; transform:translateX(-2000px) scale(0.7);} 80% {opacity:0.7; transform:translateX(0px) scale(0.7);} 100% {opacity:1; transform:scale(1);}}',
+  'back-out-up': '@keyframes back-out-up {0% {opacity:1; transform:scale(1);} 80% {opacity:0.7; transform:translateY(0px) scale(0.7);} 100% {opacity:0.7; transform:translateY(-700px) scale(0.7);}}',
+  'back-out-down': '@keyframes back-out-down {0% {opacity:1; transform:scale(1);} 80% {opacity:0.7; transform:translateY(0px) scale(0.7);} 100% {opacity:0.7; transform:translateY(700px) scale(0.7);}}',
+  'back-out-right': '@keyframes back-out-right {0% {opacity:1; transform:scale(1);} 80% {opacity:0.7; transform:translateY(0px) scale(0.7);} 100% {opacity:0.7; transform:translateX(2000px) scale(0.7);}}',
+  'back-out-left': '@keyframes back-out-left {0% {opacity:1; transform:scale(1);} 80% {opacity:0.7; transform:translateX(-2000px) scale(0.7);} 100% {opacity:0.7; transform:translateY(-700px) scale(0.7);}}',
 }
 
-// https://windicss.org/plugins/community/animations.html
 export const animations: Rule[] = [
-  [/^animate-(spin|ping|pulse|bounce)$/, ([, name], { constructCSS }) => `${keyframes[name]}\n${constructCSS({ animation: `${name} 1s linear infinite` })}`],
+  [/^animate-(.*)$/, ([, name], { constructCSS }) => {
+    const kf = keyframes[name]
+    if (kf)
+      return `${kf}\n${constructCSS({ 'animation-name': name })}`
```

Comment:
I think we could keep the duration and count by default and let them be overridden by other utils if present. So we have both out-of-box and customization capability.

Reply:
keep values on playground (count + duration)?

Reply:
added